src/domain/shared/adapters/velora.ts

```diff
@@ -0,0 +1,199 @@
+import type { Address, Hex } from 'viem'
+import { getAddress } from 'viem'
+import { base } from 'viem/chains'
+import { getTokenDecimals } from '@/features/leverage-tokens/leverageTokens.config'
+import { ETH_SENTINEL } from '@/lib/contracts/addresses'
+import { bpsToDecimalString, DEFAULT_SLIPPAGE_BPS } from './constants'
+import type { QuoteFn } from './types'
+
+export interface VeloraAdapterOptions {
+  chainId?: number
+  router: Address
+  /** Optional override for quote `fromAddress` (defaults to `router`). */
+  fromAddress?: Address
+  slippageBps?: number
+  baseUrl?: string
+}
+
+// Velora API response types - only fields we actually use
+type VeloraSwapResponse = {
+  priceRoute: {
+    srcAmount: string
+    destAmount: string
+    contractAddress: string
+    contractMethod: string
+  }
+  txParams: {
+    data: string
+  }
+}
+
+/**
+ * Create a QuoteFn adapter backed by Velora's Market API.
+ * Returns swap data and metadata needed for the redeemWithVelora contract function.
+ */
+export function createVeloraQuoteAdapter(opts: VeloraAdapterOptions): QuoteFn {
+  const {
+    chainId = base.id,
+    router,
+    fromAddress,
+    slippageBps = DEFAULT_SLIPPAGE_BPS,
+    baseUrl = 'https://api.velora.xyz',
+  } = opts
+
+  const slippage = bpsToDecimalString(slippageBps)
+
+  return async ({ inToken, outToken, amountIn, amountOut, intent }) => {
+    const url = buildQuoteUrl(baseUrl, {
+      chainId,
+      inToken,
+      outToken,
+      amountIn,
+      ...(typeof amountOut === 'bigint' ? { amountOut } : {}),
+      ...(intent ? { intent } : {}),
+      router,
+      fromAddress: (fromAddress ?? router) as Address,
+      toAddress: router as Address,
+      slippage,
+    })
+
+    const res = await fetch(url.toString(), { method: 'GET' })
+    if (!res.ok) throw new Error(`Velora quote failed: ${res.status} ${res.statusText}`)
+
+    const responseData = await res.json()
+
+    // Check for Velora API errors in the response body
+    if (responseData.error) {
+      const errorMessage = responseData.error
+      console.error('Velora error from API', { errorMessage })
+      throw new Error(`Velora quote failed: ${errorMessage}`)
+    }
+
+    const response = responseData as VeloraSwapResponse
+
+    const wantsNativeIn = inToken.toLowerCase() === ETH_SENTINEL.toLowerCase()
+    return mapVeloraResponseToQuote(response, wantsNativeIn, slippage)
+  }
+}
+
+// Internal helpers
+function buildQuoteUrl(
+  baseUrl: string,
+  params: {
+    chainId: number
+    inToken: Address
+    outToken: Address
+    amountIn: bigint
+    amountOut?: bigint
+    intent?: 'exactIn' | 'exactOut'
+    router: Address
+    fromAddress: Address
+    toAddress: Address
+    slippage: string
+  },
+): URL {
+  const url = new URL('/swap', baseUrl)
+  const normalizeToken = (token: Address) =>
+    token.toLowerCase() === ETH_SENTINEL.toLowerCase()
+      ? '0x0000000000000000000000000000000000000000'
```

Comment:
this zero address usage was specific to lifi, is this the case for velora too? is it even necessary now if we don't go thru eth?